### PR TITLE
Fix crash when launching ControlCatalog.Desktop (see #9599)

### LINKED ISSUE #9599: ControlCatalog.Desktop crashes with MarshalDirectiveException at startup
**Describe the bug**
Commit 2f3c5ef9 causes ControlCatalog.Desktop to crash with a MarshalDirectiveException when starting up. The message is:

`Cannot marshal 'parameter #2': Pointers cannot reference marshaled structures.  Use ByRef instead`

The exception is thrown in directx.Generated.cs on the line:

`base.AddMethod((GetDescDelegate)GetDesc);`

If you skip past this line in the debugger, any of the following calls will also throw the same exception:

`
base.AddMethod((CreateSwapChainForHwndDelegate)CreateSwapChainForHwnd); 
base.AddMethod((CreateSwapChainForCoreWindowDelegate)CreateSwapChainForCoreWindow); 
base.AddMethod((CreateSwapChainForCompositionDelegate)CreateSwapChainForComposition); 
base.AddMethod((GetDesc1Delegate)GetDesc1); 
`

**Proposed fix**
The issue seems to be that all of these methods take a struct as an argument that contains a `bool` field. This is not a blittable type, so the CLR tries to marshal the structs instead. A simple fix is to change two `bool` fields to `int` in DirectXStructs.cs:

DXGI_SWAP_CHAIN_DESC1: Change `bool Stereo` to `int Stereo`
DXGI_OUTPUT_DESC: Change `bool AttachedToDesktop` to `int AttachedToDesktop`

The original C struct declarations use type `BOOL`, which maps to `int`. These two fields are not referenced anywhere in the Avalonia code, so changing the datatype does not break the build.

When I make this change and rebuild Avalonia.Win32, both ControlCatalog.Desktop and ControlCatalog.NetCore appear to run successfully, and I don't see any obvious change in behavior when passing the `--dxgi` argument to ControlCatalog.NetCore.

**To Reproduce**

1. Pull the master branch
2. Build ControlCatalog.Desktop
3. Debug within Visual Studio

I tested this with Visual Studio 17.4.2

If you revert to c5e2f0ce (and delete the .gitignored Avalonia.Win32/DirectX/directx.Generated.cs if necessary), ControlCatalog.Desktop builds and runs successfully.

**Expected behavior**
ControlCatalog.Desktop should successfully

**Screenshots**
image

### PR DESCRIPTION
**Pull request by kivarsen**
## What does the pull request do?

Fixes #9599 

## What is the current behavior?

ControlCatalog.Desktop crashes on launch after a recent commit to add DXGI support

## What is the updated/expected behavior with this PR?

ControlCatalog.Desktop launches successfully

## How was the solution implemented (if it's not obvious)?

Change non-blittable bool fields to int to avoid marshalling

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Fixed issues
Fixes #9599

### CODE REVIEW

**Comment by kivarsen** [comment]
@ShadowMarker789 This PR relates to your recent Dxgi commit

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0026949-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-nightly/nuget/v3/index.json) [PRBUILDID]

File: src/Windows/Avalonia.Win32/DirectX/DirectXStructs.cs [hunk 1]
```diff
@@ -1245,7 +1245,7 @@ internal unsafe partial struct DXGI_OUTPUT_DESC
 
         internal RECT DesktopCoordinates;
 
-        internal bool AttachedToDesktop;
```
Comment (kekekeks) [inline src/Windows/Avalonia.Win32/DirectX/DirectXStructs.cs:1248]: Just add `@clr-map bool int` at the top of the file

**Review by kekekeks — APPROVED** [review]

**Comment by ShadowMarker789** [comment]
Ah yes, my mistake, thanks for the fix.

**Pull request merged**